python/cudf_polars/cudf_polars/dsl/expressions/rolling.py

```diff
@@ -210,9 +260,341 @@ def __init__(
+                [*key_nulls, plc.types.NullOrder.AFTER],
+            )
+
+        return [
+            Column(
+                plc.copying.scatter(
+                    plc.Table([tbl.columns()[0]]),
+                    order_index,
+                    plc.Table(
+                        [
+                            plc.Column.from_scalar(
+                                plc.Scalar.from_py(None, tbl.columns()[0].type()),
+                                n_rows,
+                            )
+                        ]
+                    ),
+                ).columns()[0],
+                name=name,
+                dtype=dtype,
+            )
+            for name, dtype, tbl in zip(
+                rank_out_names, rank_out_dtypes, rank_tables, strict=True
+            )
+        ]
+
+    def _split_named_expr(
+        self,
+    ) -> tuple[list[expr.NamedExpr], dict[str, list[expr.NamedExpr]]]:
+        """Split into reductions vs unary window operations."""
+        reductions: list[expr.NamedExpr] = []
+        unary_window_ops: dict[str, list[expr.NamedExpr]] = {
+            "rank": [],
+            "fill_null_with_strategy": [],
+            "cum_sum": [],
+        }
+
+        for ne in self.named_aggs:
+            v = ne.value
+            if isinstance(v, expr.UnaryFunction) and v.name in unary_window_ops:
+                unary_window_ops[v.name].append(ne)
+            else:
+                reductions.append(ne)
+        return reductions, unary_window_ops
+
+    def _build_window_order_index(
+        self,
+        by_cols: list[Column],
+        *,
+        row_id: plc.Column,
+        order_by_cols: list[Column] | None,
+        ob_desc: bool,
+        ob_nulls_last: bool,
+        reverse: bool = False,
+        value_col: plc.Column | None = None,
+        value_desc: bool = False,
+    ) -> plc.Column:
+        """Compute a stable row ordering for unary operations in a grouped context."""
+        cols: list[plc.Column] = [*(c.obj for c in by_cols)]
+        orders: list[plc.types.Order] = [*(k.order for k in by_cols)]
+        nulls: list[plc.types.NullOrder] = [*(k.null_order for k in by_cols)]
+
+        if value_col is not None:
+            # for rank(...).over(...) the ranked ("sorted") order takes precedence over order_by
+            cols.append(value_col)
+            orders.append(
+                plc.types.NullOrder.BEFORE if value_desc else plc.types.NullOrder.AFTER
+            )
+            nulls.append(
+                plc.types.NullOrder.BEFORE if value_desc else plc.types.NullOrder.AFTER
+            )
+
+        if order_by_cols:
+            base_dir = (
+                plc.types.Order.DESCENDING if ob_desc else plc.types.Order.ASCENDING
+            )
+            base_null = (
+                plc.types.NullOrder.AFTER
+                if ob_nulls_last
+                else plc.types.NullOrder.BEFORE
+            )
+
+            if reverse:
```

Comment:
Can `reverse` be incorporated in the `base_dir` and `base_null` assignment above so we can just use the `ob_dir = [base_dir] * len(order_by_cols)` like assignment?